hdbscan/tests/test_hdbscan.py

```diff
@@ -646,6 +643,33 @@ def test_hdbscan_is_sklearn_estimator():
     check_estimator(HDBSCAN)
 
 
+def test_hdbscan_graph():
+    # create a distance matrix, see test_hdbscan_distance_matrix
+    D = distance.squareform(distance.pdist(X))
+    D /= np.max(D)
+
+    threshold = stats.scoreatpercentile(D.flatten(), 50)
+
+    D[D >= threshold] = 0.0
+    D = sparse.csr_matrix(D)
+    D.eliminate_zeros()
+
+    # create cluster labels using precomputed metric
+    clusterer = HDBSCAN(metric="precomputed").fit(D)
+    labels_distance_matrix = clusterer.labels_
+
+    # create a graph from the distance matrix and transform the graph to a csr adjacency matrix
+    graph = nx.from_numpy_matrix(D.toarray())
+    adjacency_matrix = nx.adjacency_matrix(graph)
+
+    # create cluster labels using the graph metric
+    clusterer = HDBSCAN(metric="graph").fit(adjacency_matrix)
+    labels_hdbscan_graph = clusterer.labels_
+
+    assert sklearn.metrics.accuracy_score(labels_distance_matrix, labels_hdbscan_graph) == 1
```

Comment:
I'm not sure this makes sense to me. I think something like:

```suggestion
def test_hdbscan_graph():
    # create a distance matrix, see test_hdbscan_distance_matrix
    D = distance.squareform(distance.pdist(X))
    D /= np.max(D)

    threshold = stats.scoreatpercentile(D.flatten(), 50)

    D[D >= threshold] = 0.0
    D = sparse.csr_matrix(D)
    D.eliminate_zeros()

    # create cluster labels using precomputed metric
    dist_clusterer = HDBSCAN(metric="precomputed").fit(D)
    labels_distance_matrix = dist_clusterer.labels_

    # create a graph from the distance matrix and transform the graph to a csr adjacency matrix
    from hdbscan._hdbscan_reachability import sparse_mutual_reachability

    graph = sparse_mutual_rechablility(D.tolil())

    # create cluster labels using the graph metric
    graph_clusterer = HDBSCAN(metric="graph").fit(graph)
    labels_hdbscan_graph = graph_clusterer.labels_

    assert sklearn.metrics.accuracy_score(labels_distance_matrix, labels_hdbscan_graph) == 1
```

May fit better, and removes the networkx dependency. You could probably also check more things about the output.

More on this in the main review comment